<launch>
    
    <arg name="port" default="/dev/ttyACM0"/>
    <arg name="arduino_baud" default="115200"/>
    <arg name="debug" default="0"/>

    <arg name="use_own" default="true"/>
    <group if="$(arg use_own)">
    <rosparam file="$(find ebobot)/config/nodes/arduino_odom.yaml" />
    </group>

    <arg name ="arduino_connected" default="true"/>
    
    <group if="$(arg arduino_connected)">
        <node pkg="rosserial_arduino" type="serial_node.py" name="arduino_serial_node" output="screen">
            <param name="port" value="$(arg port)"/>
            <param name="baud" value="$(arg arduino_baud)"/>
        </node>
    </group>

    <node pkg="ebobot" type="motors_info_callback.py" name="motors_info_callback" output="screen" respawn="true" respawn_delay="10"> 
        <param name = "debug" value = "$(arg debug)"/>
    </node>
    <node pkg="ebobot" type="pid_setter.py" name="pid_setter">
        <param name = "path" value = "$(find ebobot)/config/scripts/pid_settings.txt"/>
    </node>
    <node pkg="ebobot" type="servos_setter.py" name="servos_setter">
        <param name = "file" value = "$(find ebobot)/config/scripts/servos.yaml"/>
    </node>
</launch>